### PR TITLE
Display correct link mode in "wifi status" output

### PR DESCRIPTION
**Pull request by rado17**
Added checks to identify and print link mode correctly.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

### CODE REVIEW

File: zephyr/src/supp_api.c [hunk 1]
```diff
@@ -371,7 +371,8 @@ int zephyr_supp_status(const struct device *dev,
 			status->ssid_len = ssid_len;
 			status->iface_mode = ssid->mode;
 			/* TODO: Derive this based on association IEs */
-			status->link_mode = WIFI_6;
+			status->link_mode = wpa_s->connection_he ? 6 :
+                        (wpa_s->connection_vht ? 5 : 4);
```
Comment (krish2718) [inline zephyr/src/supp_api.c:375]: 4 is 11n, what 3-0? Also should check for `wpa_s->connection_set` before accessing `connection_*`

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: zephyr/src/supp_api.c (hunk 1 above)
Comment (sachinthegreen) [inline zephyr/src/supp_api.c:375]: Can we use macros instead of magic numbers?

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: zephyr/src/supp_api.c (hunk 1 above)
Comment (sachinthegreen) [reply zephyr/src/supp_api.c:375]: Also is the above TODO still relevant?

Comment (krish2718) [reply zephyr/src/supp_api.c:375]: FYI, defines are here https://github.com/nrfconnect/sdk-zephyr/blob/main/include/zephyr/net/wifi.h#L202

**Review by sachinthegreen — APPROVED** [review]

File: zephyr/src/supp_api.c [hunk 2]
```diff
@@ -370,8 +370,17 @@ int zephyr_supp_status(const struct device *dev,
...
-			/* TODO: Derive this based on association IEs */
-			status->link_mode = WIFI_6;
+			if (wpa_s->connection_set == 1) {
+			status->link_mode = wpa_s->connection_he ? WIFI_6 :
+								wpa_s->connection_vht ? WIFI_5 :
+								wpa_s->connection_ht ? WIFI_4 :
+								wpa_s->connection_g ? WIFI_3 :
+								wpa_s->connection_a ? WIFI_2 :
+								wpa_s->connection_b ? WIFI_1 :
+								WIFI_LINK_MODE_UNKNOWN;
```
Comment (krish2718) [inline zephyr/src/supp_api.c:380]: Shouldn't this be WIFI_0?

**Review by krish2718 — CHANGES_REQUESTED** [review]
> Corrected the logic to check 802.11b rates.

Can you please do this in a separate commit?

File: zephyr/src/supp_api.c (hunk 2 above)
Comment (rado17) [reply zephyr/src/supp_api.c]: Corrected.

**Comment by rado17** [comment]
> > Corrected the logic to check 802.11b rates.
> 
> Can you please do this in a separate commit?

Done

File: zephyr/src/supp_api.c [hunk 3]
```diff
@@ -370,8 +370,16 @@ int zephyr_supp_status(const struct device *dev,
 			os_memcpy(status->ssid, _ssid, ssid_len);
 			status->ssid_len = ssid_len;
 			status->iface_mode = ssid->mode;
-			/* TODO: Derive this based on association IEs */
-			status->link_mode = WIFI_6;
+			if (wpa_s->connection_set == 1) {
+			status->link_mode = wpa_s->connection_he ? WIFI_6 :
```
Comment (krish2718) [inline zephyr/src/supp_api.c:374]: Indent is wrong.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: zephyr/src/supp_api.c (hunk 3 above)
Comment (rado17) [reply zephyr/src/supp_api.c]: Corrected.

**Comment by krish2718** [comment]
@rado17  please fix the conflicts in sdk-nrf PR and once it passes, we can then merge this.

**Comment by krish2718** [comment]
FYI, CI failing. https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-9000/3/testReport/junit/nrf7002dk_nrf5340_cpuapp_matter_light_bulb_sample.matter.light_bulb/debug/Build___Test_with_Twister___Twister_8_10___matter_light_bulb_sample_matter_light_bulb_debug/

**Review by sachinthegreen — APPROVED** [review]

**Review by sr1dh48r — APPROVED** [review]

**Pull request merged**